sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2147,8 +2147,18 @@ pi_result cuda_piDeviceGetInfo(pi_device device, pi_device_info param_name,
                    static_cast<uint32_t>(max_registers));
   }
 
-    // TODO: Investigate if this information is available on CUDA.
-  case PI_DEVICE_INFO_PCI_ADDRESS:
+  case PI_DEVICE_INFO_PCI_ADDRESS: {
+    constexpr size_t AddressBufferSize = 13;
+    char AddressBuffer[AddressBufferSize];
```

Comment:
I worry that we could be telling the `getInfoArray` function to copy 14 elements in case `strlen` uses the max value (13), which in turn would copy from outside `AddressBuffer`. Instead, I suggest we do something along the lines of:

```suggestion
    char AddressBuffer[AddressBufferSize + 1] = {'\0'};
```

Based on the documentation for the CUDA query, I doubt it would ever really become a problem, but I would prefer to play it safe.

Reply:
Ah, the joys of C string handling... 

Per [CUDA API docs](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g85295e7d9745ab8f0aa80dd1e172acfc): "`pciBusId` should be large enough to store 13 characters including the NULL-terminator," so allocating 14 bytes is a bit strange choice, IMO.

Now, HIP does not have such guarantees, so in their case, it indeed might be a problem even if the API stays within the spec, so the problem you raised definitely needs addressing.

I'm a bit icky with the approach of padding the buffer "just in case", though. How about the following?

```cpp
    sycl::detail::pi::assertion(
        cuDeviceGetPCIBusId(AddressBuffer, AddressBufferSize, device->get()) ==
        CUDA_SUCCESS);
    AddressBuffer[AddressBufferSize - 1] = '\0';
    sycl::detail::pi::assertion(strnlen(AddressBuffer, AddressBufferSize) > 0);
    // ^ For CUDA we can instead assert that the strnlen returns 12, since the API promises to do so
    return getInfoArray(strnlen(AddressBuffer, AddressBufferSize - 1) + 1,
                        param_value_size, param_value, param_value_size_ret,
                        AddressBuffer);
```

Reply:
Yes, that works for me too! Thank you. 😄